client/ayon_core/plugins/publish/validate_asset_docs.py

```diff
@@ -37,3 +49,142 @@ def process(self, instance):
                 "Instance \"{}\" doesn't have folder entity "
                 "set which is needed for publishing."
             ).format(instance.data["name"]))
+
+
+class RepairOverrideResolution(RepairAction):
+    """ Repair, force new resolution onto existing shot.
+    """
+    label = "Force new shot resolution."
+
+    def process(self, context, plugin):
+        values = ValidateFolderCreationResolution.get_shot_data(
+            context.data["hierarchyContext"]
+        )
+        entity_hub, entity, shot_data = values
+
+        for attrib in _RESOLUTION_ATTRIBS:
+            entity.attribs.set(attrib, shot_data[attrib])
+
+        entity_hub.commit_changes()
+
+
+class RepairIgnoreResolution(RepairAction):
+    """ Repair, disable resolution update in problematic instance(s).
+    """
+    label = "Do not update resolution."
+
+    def process(self, context, plugin):
+        create_context = context.data["create_context"]
+        for inst in get_errored_instances_from_context(context, plugin=plugin):
+            instance_id = inst.data.get("instance_id")
+            created_instance = create_context.get_instance_by_id(instance_id)
+            attr_values = created_instance.data["publish_attributes"].get(
+                "ValidateFolderCreationResolution", {})
+            attr_values["updateExistingFolderResolution"] = False
+
+        create_context.save_changes()
+
+
+class ValidateFolderCreationResolution(
+        pyblish.api.InstancePlugin,
+        AYONPyblishPluginMixin
+    ):
+    """ Validate resolution values before updating an existing folder.
+    """
+
+    label = "Validate new folder resolution"
+    order = pyblish.api.ValidatorOrder
+    families = ["shot"]
+    actions = [RepairIgnoreResolution, RepairOverrideResolution]
+
+    @classmethod
+    def get_shot_data(self, hierarchy_context):
+        """ Retrieve matching entity and shot_data from hierarchy_context.
+        """
+        project_name = tuple(hierarchy_context.keys())[0]
+        entity_hub = EntityHub(project_name)
+        entity_data = {project_name: entity_hub.project_entity}
+        entity_to_inspect = [(entity_data, hierarchy_context)]
+
+        while entity_to_inspect:
+            entity_data, data = entity_to_inspect.pop(0)
+
+            for name, value in data.items():
+                entity = entity_data.get(name)
+                child = value.get("children", None)
+
+                if entity and child:
+                    entity_children = {
+                        child.name: child
+                        for child in entity.children
+                    }
```

Comment:
Isn't this `child` variable overriding/shadowing the outer `child = value.get("children", None)`. Seems unintentional?